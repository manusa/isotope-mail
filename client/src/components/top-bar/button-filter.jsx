import React from 'react';
import {connect} from 'react-redux';
import {translate} from 'react-i18next';
import TopBarButton from './top-bar-button';
import FilterDialog from './filter-dialog';
import {messageFilterActive} from '../../selectors/application';
import mainCss from '../../styles/main.scss';
import styles from './button-filter.scss';

export class ButtonFilter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dialogVisible: false
    };
    this.handleOnToggleDialog = this.onToggleDialog.bind(this);
    this.handleOnCloseDialog = this.onCloseDialog.bind(this);
  }

  render() {
    const {t, active} = this.props;
    const {dialogVisible} = this.state;
    return <span
      className={`${styles['button-filter']} ${mainCss['mdc-menu-surface--anchor']}`}
      isotip={t('topBar.quickFilter')} isotip-position='bottom-end' isotip-size='small'
      isotip-hidden={dialogVisible.toString()}>
      <TopBarButton
        className={`${styles['button-filter--button']} ${active ? styles.active : ''}`}
        onClick={this.handleOnToggleDialog}>filter_list</TopBarButton>
      <FilterDialog visible={dialogVisible} closeFilterDialogHandler={this.handleOnCloseDialog} />
    </span>;
  }

  componentDidMount() {
    window.addEventListener('click', this.handleOnCloseDialog);
  }

  componentWillUnmount() {
    window.removeEventListener('click', this.handleOnCloseDialog);
  }

  onToggleDialog(event) {
    this.setState(previousState => ({dialogVisible: !previousState.dialogVisible}));
    event.stopPropagation();
  }

  onCloseDialog() {
    if (this.handleOnCloseDialog.disabled !== true) {
      this.setState({dialogVisible: false});
    }
    delete this.handleOnCloseDialog.disabled;
  }
}

const mapStateToProps = state => ({
  active: messageFilterActive(state)
});

export default connect(mapStateToProps)(translate()(ButtonFilter));
