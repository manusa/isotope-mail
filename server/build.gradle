buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
