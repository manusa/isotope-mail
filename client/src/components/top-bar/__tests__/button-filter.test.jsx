import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ButtonFilter} from '../button-filter';
import MessageFilters from '../../../services/message-filters';

describe('ButtonFilter component test suite', () => {
  describe('Snapshot render', () => {
    test('Snapshot render, not active, should render deactivated button-filter', () => {
      // Given
      const props = {t: jest.fn(arg => arg), active: false};
      // When
      const buttonFilter = shallow(<ButtonFilter {...props}/>);
      // Then
      expect(buttonFilter).toMatchSnapshot();
    });
    test('Snapshot render, active, should render activated button-filter', () => {
      // Given
      const props = {t: jest.fn(arg => arg), active: true};
      // When
      const buttonFilter = shallow(<ButtonFilter {...props}/>);
      // Then
      expect(buttonFilter).toMatchSnapshot();
    });
  });
  describe('Events tests', () => {
    test('unmount, window should not contain event listeners', () => {
      // Given
      const originalAddEventListener = window.addEventListener;
      const originalRemoveEventListener = window.removeEventListener;
      window.addEventListener = jest.fn(() => originalAddEventListener.apply(null, arguments));
      window.removeEventListener = jest.fn(() => originalRemoveEventListener.apply(null, arguments));
      const props = {t: jest.fn(arg => arg), activeMessageFilter: MessageFilters.READ};
      const buttonFilter = shallow(<ButtonFilter {...props}/>);
      // When
      buttonFilter.unmount();
      // Then
      expect(window.addEventListener).toHaveBeenCalledTimes(1);
      expect(window.removeEventListener).toHaveBeenCalledTimes(1);
    });
    test('onToggleDialog, state is changed', () => {
      // Given
      const props = {t: jest.fn(arg => arg), active: true};
      const buttonFilter = shallow(<ButtonFilter {...props}/>);
      const event = {stopPropagation: jest.fn()};
      // When
      buttonFilter.find('TopBarButton').props().onClick(event);
      // Then
      expect(event.stopPropagation).toHaveBeenCalledTimes(1);
      expect(buttonFilter.state().dialogVisible).toBe(true);
    });
    test('onCloseDialog, dialog is visible and function is NOT disabled, state.dialogVisible is false', () => {
      // Given
      const props = {t: jest.fn(arg => arg), active: true};
      const buttonFilter = shallow(<ButtonFilter {...props}/>);
      buttonFilter.state().dialogVisible = true;
      // When
      buttonFilter.instance().handleOnCloseDialog();
      // Then
      expect(buttonFilter.state().dialogVisible).toBe(false);
    });
    test('onCloseDialog, disalog is visible and function is disabled, state.dialogVisible is true', () => {
      // Given
      const props = {t: jest.fn(arg => arg), active: true};
      const buttonFilter = shallow(<ButtonFilter {...props}/>);
      buttonFilter.state().dialogVisible = true;
      buttonFilter.find('Connect(Translate(FilterDialog))').props().closeFilterDialogHandler.disabled = true;
      // When
      buttonFilter.instance().handleOnCloseDialog();
      // Then
      expect(buttonFilter.state().dialogVisible).toBe(true);
    });
  });
});
