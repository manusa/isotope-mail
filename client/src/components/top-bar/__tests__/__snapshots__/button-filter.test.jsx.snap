// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonFilter component test suite Snapshot render Snapshot render, active, should render activated button-filter 1`] = `
<span
  className="button-filter mdc-menu-surface--anchor"
  isotip="topBar.quickFilter"
  isotip-hidden="false"
  isotip-position="bottom-end"
  isotip-size="small"
>
  <TopBarButton
    className="button-filter--button active"
    onClick={[Function]}
  >
    filter_list
  </TopBarButton>
  <Connect(Translate(FilterDialog))
    closeFilterDialogHandler={[Function]}
    visible={false}
  />
</span>
`;

exports[`ButtonFilter component test suite Snapshot render Snapshot render, activeMessageFilter=ALL, should render deactivated button-filter 1`] = `
<span
  className="button-filter mdc-menu-surface--anchor"
  isotip="topBar.quickFilter"
  isotip-hidden="false"
  isotip-position="bottom-end"
  isotip-size="small"
>
  <TopBarButton
    className="button-filter--button "
    onClick={[Function]}
  >
    filter_list
  </TopBarButton>
  <Connect(Translate(FilterDialog))
    closeFilterDialogHandler={[Function]}
    visible={false}
  />
</span>
`;

exports[`ButtonFilter component test suite Snapshot render Snapshot render, activeMessageFilter=READ, should render activated button-filter 1`] = `
<span
  className="button-filter mdc-menu-surface--anchor"
  isotip="topBar.quickFilter"
  isotip-hidden="false"
  isotip-position="bottom-end"
  isotip-size="small"
>
  <TopBarButton
    className="button-filter--button active"
    onClick={[Function]}
  >
    filter_list
  </TopBarButton>
  <Connect(Translate(FilterDialog))
    visible={false}
  />
</span>
`;

exports[`ButtonFilter component test suite Snapshot render Snapshot render, not active, should render deactivated button-filter 1`] = `
<span
  className="button-filter mdc-menu-surface--anchor"
  isotip="topBar.quickFilter"
  isotip-hidden="false"
  isotip-position="bottom-end"
  isotip-size="small"
>
  <TopBarButton
    className="button-filter--button "
    onClick={[Function]}
  >
    filter_list
  </TopBarButton>
  <Connect(Translate(FilterDialog))
    closeFilterDialogHandler={[Function]}
    visible={false}
  />
</span>
`;
